src/app/auth/login/login.component.ts

```diff
@@ -27,36 +33,150 @@ export class LoginComponent {
     this.signIn = true;
   }
 
+  public get loginEmailValid() {
+    return (
+      this.loginForm.get('emailLogin')!.hasError('pattern') &&
+      !this.loginForm.get('emailLogin')!.hasError('required')
+    );
+  }
+
```

Comment:
loginEmailValid() and loginPasswordlValid() are similar. 
Could be simplify by creation an one method for both checks.